service/k8sclient/k8sclient_test.go

```diff
@@ -39,6 +42,143 @@ import (
 	"github.com/percona-platform/dbaas-controller/utils/logger"
 )
 
+// VersionServiceClient represents a client for Version Service API.
+type VersionServiceClient struct {
+	url  string
+	http *http.Client
+}
+
+// NewVersionServiceClient creates a new client for given version service URL.
+func NewVersionServiceClient(url string) *VersionServiceClient {
+	return &VersionServiceClient{
+		url: url,
+		http: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+	}
+}
+
+// componentsParams contains params to filter components in version service API.
+type componentsParams struct {
+	product        string
+	productVersion string
+}
+
+type matrix struct {
+	PXCOperator   map[string]interface{} `json:"pxcOperator,omitempty"`
+	PSMDBOperator map[string]interface{} `json:"psmdbOperator,omitempty"`
+}
+
+type Version struct {
+	Product        string `json:"product"`
+	ProductVersion string `json:"operator"`
+	Matrix         matrix `json:"matrix"`
+}
+
+// VersionServiceResponse represents response from version service API.
+type VersionServiceResponse struct {
+	Versions []Version `json:"versions"`
+}
+
+var errNoVersionsFound error = errors.New("no versions to compare current version with found")
+
+func latest(m map[string]interface{}) (*goversion.Version, error) {
+	if len(m) == 0 {
+		return nil, errNoVersionsFound
+	}
+	latest := goversion.Must(goversion.NewVersion("0.0.0"))
+	for version := range m {
+		parsedVersion, err := goversion.NewVersion(version)
+		if err != nil {
+			return nil, err
+		}
+		if parsedVersion.GreaterThan(latest) {
+			latest = parsedVersion
+		}
+	}
+	return latest, nil
+}
+
+func latestProduct(s []Version) (*goversion.Version, error) {
```

Comment:
is that func used somewhere else except the test?

Reply:
It is used only in tests.

Reply:
everything is good 👍